### PR TITLE
my homework(3)

### PR DESCRIPTION
**Pull request by MihailNenahov**
not finished. need help.

### CODE REVIEW

**Comment by MihailNenahov** [comment]
now almost everything is ok except avarage. i left f comment.

**Comment by MihailNenahov** [comment]
After updating the task. I can say, that task completed absolutely.

File: third_task/max/max.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		"dog",
+		"cat",
+		"string",
+		"aaaaaa",
+	}
+	fmt.Println(max(mySlice))
+}
+
+func max(givenSlice []string) (res string) {
+	res = givenSlice[0]
```
Comment (imatsiash-ib) [inline third_task/max/max.go:24]: What happens if `givenSlice` is empty slice ? I guess panic will happen here.

File: third_task/max/max.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	for _, tmp := range givenSlice {
+		if len(res) < len(tmp) {
```
Comment (imatsiash-ib) [inline third_task/max/max.go]: Please use `utf8.RuneCountInString` function to get length of utf-8 string.

File: third_task/printSorted/printSorted.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		keys = append(keys, k)
+	}
+
+	for i := 0; i < len(keys); i++ {
+		for j := i + 1; j < len(keys); j++ {
+			if keys[i] > keys[j] {
+				keys[i], keys[j] = keys[j], keys[i]
+			}
+		}
+	}
```
Comment (imatsiash-ib) [inline third_task/printSorted/printSorted.go]: It is ok. You can also use https://golang.org/pkg/sort/ package to sort slice.

File: third_task/printSorted/printSorted.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+	fmt.Println(keys)
+
+	for _, k := range keys {
+		fmt.Println(givenMap[k])
+		res = append(res, givenMap[k])
```
Comment (imatsiash-ib) [inline third_task/printSorted/printSorted.go]: Do we need this map ? According to the task description function should just print elements.

File: third_task/printSorted/printSorted.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+		keys = append(keys, k)
+	}
+
+	sort.Ints(keys)
+
+	for _, k := range keys {
+		res = append(res, givenMap[k])
+	}
+
+	fmt.Println(res)
```
Comment (imatsiash-ib) [inline third_task/printSorted/printSorted.go]: We should print map values, not keys.

Comment (MihailNenahov) [reply third_task/printSorted/printSorted.go]: everything is correct, but variable has incorrect name.

**Review by imatsiash-ib — APPROVED** [review]

File: third_task/average/average.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+func main() {
+	myArray := [5]int{1, 2, 3, 4, 5}
+	fmt.Println(average(myArray[:]))
+	myArray2 := [6]int{1, 2, 3, 4, 5, 6}
+	fmt.Println(average(myArray2[:]))
+}
+
+func average(givenArray []int) float64 {
+	var Res int
```
Comment (burov) [inline third_task/average/average.go]: local function names, should start from lower case letter

File: third_task/printSorted/printSorted.go (hunk 5 above)
Comment (burov) [reply third_task/printSorted/printSorted.go]: Creating new slice for this purpose not a good idea

File: third_task/reverse/reverse.go [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import "fmt"
+
+func main() {
+	mySlice := []int{1, 2, 3, 4, 5}
+	fmt.Println(reverse(mySlice))
+	newSlice := reverse(mySlice)
```
Comment (burov) [inline third_task/reverse/reverse.go]: Why function there called twice ?

**Review by burov — APPROVED** [review]
Overall looks ok, but a few improvements possible

File: third_task/reverse/reverse.go (hunk 7 above)
Comment (MihailNenahov) [reply third_task/reverse/reverse.go]: i checked, that func returns right result and can init some variable. now this code is not neсessary.

**Comment by MihailNenahov** [comment]
now in this request we also have task from 4th lecture. u need to check only last commit.

**Pull request merged**